src/core/utils/spec-preprocess.ts

```diff
@@ -261,6 +262,19 @@ export function traverseToFixSpecDownstream(spec: GoslingSpec | SingleView, pare
             if (Is2DTrack(track)) {
                 // TODO: Add a test for this.
                 track.layout = 'linear';
+
+                /**
+                 * Add y-axis domain
+                 */
+                if ((IsSingleTrack(track) || IsOverlaidTrack(track)) && IsChannelDeep(track.y) && !track.y.domain) {
```

Comment:
The `yDomain` value is overridden by children tracks here (i.e., `y: { ..., domain: {...} }`).